src/main/c/ssl.c

```diff
@@ -1178,11 +1178,11 @@ TCN_IMPLEMENT_CALL(jint, SSL, getLastErrorNumber)(TCN_STDARGS) {
 }
 
 static void ssl_info_callback(const SSL *ssl, int where, int ret) {
-    tcn_ssl_ctxt_t *c;
+    uint16_t *handshakeCount = NULL;
     if (0 != (where & SSL_CB_HANDSHAKE_START)) {
-        c = SSL_get_app_data2(ssl);
-        if (c != NULL) {
-            ++c->handshakeCount;
+        handshakeCount = (uint16_t*) SSL_get_app_data3(ssl);
+        if (handshakeCount != NULL) {
+            ++(*handshakeCount);
```

Comment:
@Scottmitch btw I think I could remove the `()` around it but I kind of think it is easier to read with them. What you prefer ?